### PR TITLE
[tests-only][full-ci] Backport test PRs

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Backport PRs:
- #5365 
- #5391 
- #5417 
- #5424 
- #5462 
- #5492
- #5529 
- #5536
- #5537 
- #5541 
- #5545
- #5562 
- #5575 
- #5579
- #5629 
- #5605
- #5606 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by amrita-shrestha** [comment]
@saw-jan Assign role via graph is added in oCIS master only. Please visit this link for verification https://github.com/owncloud/web/issues/8325 
I think there is no need of  backporting this PR https://github.com/owncloud/ocis/pull/5579 and https://github.com/owncloud/ocis/pull/5629

**Comment by saw-jan** [comment]
> @saw-jan Assign role via graph is added in oCIS master only. Please visit this link for verification [owncloud/web#8325](https://github.com/owncloud/web/issues/8325) I think there is no need of backporting this PR #5579 and #5629

We can make the differences feature files
But the php codes need to be almost similar

**Comment by saw-jan** [comment]
@grgprarup Please, review whether I have brought back the tests/codes that you have cleaned

**Comment by amrita-shrestha** [comment]
> > @saw-jan Assign role via graph is added in oCIS master only. Please visit this link for verification [owncloud/web#8325](https://github.com/owncloud/web/issues/8325) I think there is no need of backporting this PR #5579 and #5629
> 
> We can make the differences in feature files But the php codes need to be almost similar. Making them similar makes it easier for future porting as well

If we not going to use such php code then does it even create problems in backporting?

**Review by grgprarup — COMMENTED** [review]
lgtm for me.

**Comment by saw-jan** [comment]
> If we not going to use such php code then does it even create problems in backporting?

There should not be any issue if the file changes are in different lines. But mostly we will change Graph and Space related files so that will increase the chances of conflicts

File: tests/TestHelpers/GraphHelper.php [hunk 1]
```diff
@@ -478,34 +479,23 @@ public static function deleteGroup(
...
 	 * @param string $adminPassword
 	 * @param string $groupId
-	 * @param array $users expects users array with user ids
-	 *                     [ [ 'id' => 'some_id' ], ]
 	 *
 	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public static function addUsersToGroup(
+	public static function deleteGroup(
```
Comment (amrita-shrestha) [inline tests/TestHelpers/GraphHelper.php:486]: Now I know from which PR this method removed 😄

File: tests/acceptance/features/apiSpaces/changeSpaces.feature [hunk 2]
```diff
@@ -14,8 +14,12 @@ Feature: Change data of space
       | Bob      |
     And the administrator has given "Alice" the role "Space Admin" using the settings api
     And user "Alice" has created a space "Project Jupiter" of type "project" with quota "20"
-    And user "Alice" has shared a space "Project Jupiter" to user "Brian" with role "editor"
-    And user "Alice" has shared a space "Project Jupiter" to user "Bob" with role "viewer"
+    And user "Alice" has shared a space "Project Jupiter" with settings:
+      | shareWith | Brian    |
+      | role      | editor |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/changeSpaces.feature:19]:
```suggestion
      | shareWith | Brian    |
      | role      | editor   |
```

File: tests/acceptance/features/apiSpacesShares/shareSubItemOfSpace.feature [hunk 3]
```diff
@@ -18,43 +18,62 @@ Feature: Share a file or folder that is inside a space
...
     And user "Alice" has created a folder "folder" in space "share sub-item"
     And user "Alice" has uploaded a file inside space "share sub-item" with content "some content" to "file.txt"
-  And using new DAV path
+    And using new DAV path
+
 
-    
   Scenario Outline: A manager of the space can share an entity inside project space to another user with role
-    When user "Alice" shares the following entity "<entity>" inside of space "share sub-item" with user "Brian" with role "<role>"
+    And user "Alice" creates a share inside of space "share sub-item" with settings:
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpacesShares/shareSubItemOfSpace.feature:25]:
```suggestion
    When user "Alice" creates a share inside of space "share sub-item" with settings:
```

File: tests/acceptance/features/apiSpacesShares/shareSubItemOfSpace.feature [hunk 4]
```diff
@@ -18,43 +18,62 @@ Feature: Share a file or folder that is inside a space
...
-      | entity   | type   | role   |
-      | folder   | folder | viewer |
-      | folder   | folder | editor |
-      | file.txt | file   | viewer |
-      | file.txt | file   | editor |
+      | entity   | type   | role   | expireDate               | expiration |
+      | folder   | folder | viewer |                          |            |
+      | folder   | folder | editor |                          |            |
+      | file.txt | file   | viewer |                          |            |
+      | file.txt | file   | editor |                          |            |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpacesShares/shareSubItemOfSpace.feature:42]:
```suggestion
      | entity   | type   | role   | 
      | folder   | folder | viewer |  
      | file.txt | file   | viewer |         
      | file.txt | file   | editor |      
```

**Review by amrita-shrestha — COMMENTED** [review]
i know you just cherry-pick other pick but i think there is some place where we can improve

**Review by amrita-shrestha — CHANGES_REQUESTED** [review]
> As we discussed today with @ScharfViktor and @individual-it, we will maintain similar test (php) codes and feature files as well - and we tag skipOnStable2.0 to the scenarios that are not implemented in the stable branch.

> Key points:

>  - Maintain possible similar php codes
>   - tag scenarios that are not implemented in stable branch with skipOnStable2.0

This PR has not added scenario with skipOnStable2.0 tag . Please add that also

File: tests/TestHelpers/GraphHelper.php [hunk 5]
```diff
@@ -451,24 +451,25 @@ public static function getGroups(
...
 	 * @param string $adminUser
 	 * @param string $adminPassword
-	 * @param string $groupId
+	 * @param string $groupName
 	 *
 	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public static function deleteGroup(
+	public static function getGroup(
```
[the remark this answers is not in the document]
Comment (grgprarup) [reply tests/TestHelpers/GraphHelper.php:459]: `deleteGroup` is not deleted, just line number is changed.

**Comment by individual-it** [comment]
> i know you just cherry-pick other commit but i think there is some place where we can improve

I would suggest to have those small improvements in a separate PR, because if you do them in this PR, you have to port them again into master, so with a separate PR it would be cleaner and easier

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by amrita-shrestha — APPROVED** [review]
lgtm

**Review by ScharfViktor — APPROVED** [review]
LGTM :thumbsup:

File: tests/acceptance/features/apiSpaces/spaceManagement.feature [hunk 6]
```diff
@@ -0,0 +1,171 @@
+@api @skipOnOcV10
```
Comment (phil-davis) [inline tests/acceptance/features/apiSpaces/spaceManagement.feature:1]: I thought that we already deleted all the `skipOnOcV10` tags.
I will do that as a separate PR.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
The docker-arm pipeline https://drone.owncloud.com/owncloud/ocis/19870/85 does not start because there are no drone agents for that any more.

I will have a look and probably this PR just needs a rebase against the latest `stable-2.0`

**Pull request merged**